src/main/java/de/tum/in/www1/artemis/web/rest/TextAssessmentEventResource.java

```diff
@@ -0,0 +1,108 @@
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.analytics.TextAssessmentEvent;
+import de.tum.in.www1.artemis.repository.CourseRepository;
+import de.tum.in.www1.artemis.repository.TextAssessmentEventRepository;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+import de.tum.in.www1.artemis.web.rest.errors.EntityNotFoundException;
+
+/**
+ * REST controller for managing TextAssessmentEventResource.
+ */
+@RestController
+@RequestMapping("/analytics/text-assessment-events")
+public class TextAssessmentEventResource {
+
+    private final Logger log = LoggerFactory.getLogger(TextAssessmentEventResource.class);
+
+    private final TextAssessmentEventRepository textAssessmentEventRepository;
+
+    private final AuthorizationCheckService authCheckService;
+
+    private final UserRepository userRepository;
+
+    private final CourseRepository courseRepository;
+
+    @Value("${info.text-assessment-analytics-enabled}")
+    private Optional<Boolean> textAssessmentAnalyticsEnabled;
+
+    public TextAssessmentEventResource(TextAssessmentEventRepository textAssessmentEventRepository, AuthorizationCheckService authCheckService, UserRepository userRepository,
+            CourseRepository courseRepository) {
+        this.textAssessmentEventRepository = textAssessmentEventRepository;
+        this.authCheckService = authCheckService;
+        this.userRepository = userRepository;
+        this.courseRepository = courseRepository;
+    }
+
+    /**
+     * The text assessment analytics are enabled when the configuration info.text-assessment-analytics-enabled is set to true.
+     * A non existing entry or false mean that the text assessment analytics is not enabled
+     * @return whether the text assessment analytics are enabled or not
+     */
+    private boolean isTextAssessmentAnalyticsEnabled() {
+        return textAssessmentAnalyticsEnabled.isPresent() && Boolean.TRUE.equals(textAssessmentAnalyticsEnabled.get());
+    }
+
+    /**
+     * This function retrieves all of the events from the 'text_assessment_event' table
+     * @return returns a List of TextAssessmentEvent's
+     */
+    @GetMapping("/all")
+    @PreAuthorize("hasRole('ADMIN')")
+    public ResponseEntity<List<TextAssessmentEvent>> getAllEvents() {
+        List<TextAssessmentEvent> events = textAssessmentEventRepository.findAll();
+        return ResponseEntity.ok().body(events);
+    }
+
+    /**
+     * This function adds an assessment event into the text_assessment_event table.
+     * @param event to be added
+     * @return the status of the finished request
+     * @throws URISyntaxException
+     */
+    @PostMapping("/add")
```

Comment:
still the wrong REST URL, POST also works on the entity URL in plural, e.g.

@PostMapping("/courses") would allow you to create a new course

Reply:
I took this one as a reference for these two:

https://github.com/ls1intum/Artemis/blob/develop/src/main/java/de/tum/in/www1/artemis/web/rest/OrganizationResource.java#L182

and this:

https://github.com/ls1intum/Artemis/blob/develop/src/main/java/de/tum/in/www1/artemis/web/rest/OrganizationResource.java#L134

Reply:
but I guess that's not a good reference point. I will change to as suggested. 👍